diff --git a/deps/CMakeLists.txt b/deps/CMakeLists.txt
index 10abf8e..cbd13e5 100644
--- a/deps/CMakeLists.txt
+++ b/deps/CMakeLists.txt
@@ -154,11 +154,11 @@ if (NOT TARGET nanoflann)
     target_include_directories(nanoflann INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/nanoflann/include>)
 
     # For backward compatibility, we need to make a symlink in the build directory.
-    file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/nanoflann)
-    file(CREATE_LINK
-        ${CMAKE_CURRENT_SOURCE_DIR}/nanoflann/include/nanoflann.hpp
-        ${CMAKE_CURRENT_BINARY_DIR}/include/nanoflann/nanoflann.hpp SYMBOLIC)
-    target_include_directories(nanoflann INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>)
+    # file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/nanoflann)
+    # file(CREATE_LINK
+    #     ${CMAKE_CURRENT_SOURCE_DIR}/nanoflann/include/nanoflann.hpp
+    #     ${CMAKE_CURRENT_BINARY_DIR}/include/nanoflann/nanoflann.hpp SYMBOLIC)
+    # target_include_directories(nanoflann INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>)
 endif()
 
 if (NOT TARGET happly)
